erpnext/manufacturing/doctype/bom/bom.js

```diff
@@ -734,3 +740,44 @@ function trigger_process_loss_qty_prompt(frm, cdt, cdn, item_code) {
 		__("Set Quantity")
 	);
 }
+
+function add_scrap_item(frm, cdt, cdn) {
+	var item = locals[cdt][cdn];
+
+	if (item.scrap) {
+		let row = null;
+		var scrap_item_qty = (item.qty / 100) * item.scrap;
+
+		for (var i = 0; i < frm.doc.scrap_items.length; i++) {
+			if (frm.doc.scrap_items[i].item_code == item.item_code) {
+				row = frm.doc.scrap_items[i];
+				break;
+			}
+		}
+
+		if (row) {
+			row.stock_qty = scrap_item_qty;
+			row.rate = item.rate;
+			row.amount = scrap_item_qty * item.rate;
+			row.stock_uom = item.stock_uom;
+			row.base_rate = item.base_rate;
+			row.base_amount = item.base_rate * scrap_item_qty;
```

Comment:
Do we need to update these ? This function is triggered only on changing scrap %, so only qty and related stuff like amount needs a change right ? why would the rate change?

Reply:
Yes, In some scenarios

Scenario: 
1. Add an item in the Items.
2. Set "Scrap %" (will populate a new row in Scrap)
3. Again add the same item with a different qty or rate and set "Scrap %".

Now, the existing row in Scrap will be updated with new values i.e qty and rate.